src/audio/selector/selector.c

```diff
@@ -83,6 +91,140 @@ static struct comp_dev *selector_new(const struct comp_driver *drv,
 	return dev;
 }
 
+static void set_selector_params(struct comp_dev *dev,
+				struct sof_ipc_stream_params *params)
+{
+}
+#else
+static void build_config(struct comp_data *cd)
+{
+	enum sof_ipc_frame valid_format;
+
+	cd->source_format = cd->md.base_cfg.audio_fmt.depth;
+	audio_stream_fmt_conversion(cd->md.base_cfg.audio_fmt.depth,
+				    cd->md.base_cfg.audio_fmt.valid_bit_depth,
+				    &cd->source_format,
+				    &valid_format,
+				    cd->md.base_cfg.audio_fmt.s_type);
+
+	audio_stream_fmt_conversion(cd->md.output_format.depth,
+				    cd->md.output_format.valid_bit_depth,
+				    &cd->sink_format,
+				    &valid_format,
+				    cd->md.output_format.s_type);
+
+	cd->config.in_channels_count = cd->md.base_cfg.audio_fmt.channels_count;
+	cd->config.out_channels_count = cd->md.output_format.channels_count;
+}
+
+static struct comp_dev *selector_new(const struct comp_driver *drv,
```

Comment:
This should use the module API now

Reply:
@lgirdwood  this is my step two since we can validate it first on windows

Reply:
that's fine, but can step 2 be part of this PR (as the second commit perhaps). I really dont want to take new processing that does not use the module API.

Reply:
sure. I will work on it

Reply:
And we also get a issue https://github.com/thesofproject/sof/issues/5986. I found a bug in our module adapter framework

Reply:
@ranj063 can you advise here ?

Reply:
@lgirdwood I have a tentative fix for the first part in https://github.com/thhttps://github.com/thesofproject/sof/issues/5986#issuecomment-1176691826

But I also commented that there's a fundamental problem with the module binding code in the IPC4 helper where we access the module's private data in the common code. I don't have an elegant fix for it yet.

Reply:
here's the PR for the first part to get module loading to work: https://github.com/thesofproject/sof/pull/5994

Reply:
@ranj063 will be worth syncing with @softwarecki @marcinszkudlinski and @abonislawski here for best IPC4 solution.

Reply:
@ranj063 @lgirdwood In my latest PR i introduce a new function to get access to IPC4 base module configuration. This function still access a private components data. Use this new function will make easier solve this issue in the future.
https://github.com/thesofproject/sof/pull/5985/commits/1dc40945385728bc5826805f21b0b5e5dc5151ee

Reply:
@RanderWang @ranj063 I've merged the PR patch series from @softwarecki now that has improvements for IPC4 access.